app/src/androidTest/java/org/kiwix/kiwixmobile/main/KiwixMainRobot.kt

```diff
@@ -0,0 +1,158 @@
+/*
+ * Kiwix Android
+ * Copyright (c) 2019 Kiwix <android.kiwix.org>
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+package org.kiwix.kiwixmobile.main
+
+import applyWithViewHierarchyPrinting
+import org.kiwix.kiwixmobile.BaseRobot
+import org.kiwix.kiwixmobile.Findable.StringId.ContentDesc
+import org.kiwix.kiwixmobile.Findable.Text
+import org.kiwix.kiwixmobile.Findable.ViewId
+import org.kiwix.kiwixmobile.R
+import org.kiwix.kiwixmobile.testutils.TestUtils.getResourceString
+import org.kiwix.kiwixmobile.utils.StandardActions.openDrawer
+
+fun kiwixMainRobo(func: KiwixMainRobot.() -> Unit) =
+  KiwixMainRobot().applyWithViewHierarchyPrinting(func)
+
+class KiwixMainRobot : BaseRobot() {
+
+  init {
+    isVisible(ViewId(R.id.navigation_container))
+  }
+
+  fun clickReaderOnBottomNav() {
+    clickOn(ViewId(R.id.readerFragment))
+  }
+
+  fun assertReaderScreenDisplayed() {
+    isVisible(Text(getResourceString(R.string.reader)))
+  }
+
+  fun clickLibraryOnBottomNav() {
+    clickOn(ViewId(R.id.libraryFragment))
+  }
+
+  fun library(func: LibraryRobot.() -> Unit) =
+    LibraryRobot().applyWithViewHierarchyPrinting(func)
+
+  inner class LibraryRobot : BaseRobot() {
+    init {
+      isVisible(Text(getResourceString(R.string.library)))
```

Comment:
this is not a UI element unique to the Library screen so this assertion is not reliable for confirming "Yes you are on this screen and it safe to use these functions"

Reply:
This is the case with the Reader screen I believe but not with this, as I don't think there is _Library_ written at any different screen on a toolbar

Reply:
Library is on every top level destination because of the nav bar

Reply:
But the textView "Library" isn't visible anywhere else na.

Reply:
This scans the screen for any occurrence of the text "Library". If I am on the Download screen this assertion still passes because of the NavBar, it is not proving we are on Library so it is not fit for purpose